Add plugin: AI Transcriber

mssoftjp opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:https://github.com/mssoftjp/obsidian-ai-transcriber

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

mssoftjp commented:
I've now changed the repository from private to public. Could you  please check again?

mssoftjp commented:
@Zachatoo Thanks for the quick review—and sorry for the misses. I’ll carefully address all the points you raised and push an updated build.

Zachatoo commented:
[this.animationInterval = this.registerInterval](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/core/utils/LoadingAnimation.ts#L50-L52)
`this.registerInterval()` should always be provided, so your plugin doesn't cause a memory leak.
I recommend you update your `LoadingAnimation` class to extend the [`Component`](https://docs.obsidian.md/Reference/TypeScript+API/Component) class, and everywhere where you use the `LoadingAnimation` class you use [`this.addChild(new LoadingAnimation())`](https://docs.obsidian.md/Reference/TypeScript+API/Component/addChild) so it's parent can manage the lifecycle of the `LoadingAnimation` component, ensuring it doesn't register an interval for longer than it's needed.

[this.updateInterval = this.plugin.registerInterval(](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/ui/TranscriptionView.ts#L73), [if (this.updateInterval !== null) {](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/ui/TranscriptionView.ts#L88-L91)
Please use `this.registerInterval()` instead, so the interval is unloaded automatically when the view is unloaded, and don't manage clearing it yourself.

[this.app.workspace.detachLeavesOfType(VIEW_TYPE_TRANSCRIPTION);](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/main-api.ts#L162)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[await this.app.vault.process(createdFile, () => formattedTranscription);](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/ui/ApiTranscriptionModal.ts#L793), [await this.app.vault.process(createdFile, () => formattedTranscription);](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/ui/ApiTranscriptionModal.ts#L866)
[`Vault.process`](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+%60Vault.process%60+instead+of+%60Vault.modify%60+to+modify+a+file+in+the+background) should not be used in this way. Please refer to [this YouTube video](https://youtu.be/RvtV9DKr3ag?t=52) for how the process API works and how to properly use it.

[for (const part of parts) {](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/1cf4a8e4b8c68f18d43f587b0f7426e7019e5c19/src/ui/ApiTranscriptionModal.ts#L964-L975)
`createFolder()` will automatically create all missing parts of the folder path for you, you don't need to iterate over each part of the path and create the folder yourself.

mssoftjp commented:
Thank you, @Zachatoo  — your feedback was very helpful. I went through your notes in order and made the following changes.

1) Loading animation lifecycle
I refactored `LoadingAnimation` to extend `Component`, and it now always registers its interval via `Component#registerInterval`. I also updated call sites to instantiate it via `addChild(new LoadingAnimation())`, so the parent component manages its lifecycle.
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/core/utils/LoadingAnimation.ts#L1-L48
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/TranscriptionView.ts#L35-L40
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/StatusBarManager.ts#L19-L23
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/ApiTranscriptionModal.ts#L59-L67

2) View interval registration
In `TranscriptionView`, I switched to `this.registerInterval(window.setInterval(...))` so the interval is automatically cancelled when the view unloads, and removed the manual clearing logic.
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/TranscriptionView.ts#L54-L99

3) Don’t detach leaves in `onunload`
I removed `detachLeavesOfType(...)` from plugin cleanup as per the guidelines.
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/main-api.ts#L158-L173

4) `Vault.process` usage
I removed the “write a constant string” usage of `Vault.process` and replaced it with `Vault.modify(createdFile, formattedTranscription)` since we weren’t doing a read-modify-write based on `currentContent`. I kept `Vault.process(file, (currentContent) => ...)` only where we actually depend on the existing content (append fallback).
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/ApiTranscriptionModal.ts#L748-L902

5) Folder creation
I simplified folder creation to a single `createFolder(normalizePath(folderPath))` call (with an existence/type check).
https://github.com/mssoftjp/obsidian-ai-transcriber/blob/4fc7d03be0294cd5c5e6a30afd9ac6d365f3a886/src/ui/ApiTranscriptionModal.ts#L957-L973
